packages/compass-telemetry/src/telemetry-events.ts

```diff
@@ -0,0 +1,889 @@
+  payload: { id: any };
+};
+
+type SignalOpenedEvent = {
+  name: 'Signal Opened';
+  payload: { id: any };
+};
+
+type SignalActionButtonClickedEvent = {
+  name: 'Signal Action Button Clicked';
+  payload: { id: any };
+};
+
+type SignalLinkClickedEvent = {
+  name: 'Signal Link Clicked';
+  payload: { id: any };
+};
+
+type SignalClosedEvent = {
+  name: 'Signal Closed';
+  payload: { id: any };
+};
+
+type StringEvent = {
+  name: 'string';
+  payload: { value: number };
+};
+
+type ConnectionAttemptEvent = ConnectionScoped<{
+  name: 'Connection Attempt';
+  payload: { is_favorite: boolean; is_recent: boolean; is_new: boolean };
+}>;
+
+type NewConnectionEvent = ConnectionScoped<{
+  name: 'New Connection';
+  payload: {
+    is_atlas: boolean;
+    atlas_hostname: string;
+    is_local_atlas: boolean;
+    is_dataLake: boolean;
+    is_enterprise: boolean;
+    is_genuine: boolean;
+    non_genuine_server_name: string;
+    server_version: string;
+    server_arch: string;
+    server_os_family: string;
+    topology_type: string;
+  };
+}>;
+
+type ConnectionFailedEvent = ConnectionScoped<{
+  name: 'Connection Failed';
+  payload: { error_code: string | number; error_name: string };
+}>;
+
+type ApplicationLaunchedEvent = {
+  name: 'Application Launched';
+  payload: {
+    context: string;
+    launch_connection: string;
+    protected: boolean;
+    readOnly: boolean;
+    maxTimeMS: number;
+    global_config: boolean;
+    cli_args: boolean;
+    legacy_connections: boolean;
+  };
+};
+
+type AutoupdateAcceptedEvent = {
+  name: 'Autoupdate Accepted';
+  payload: {
+    update_version?: string;
+    manual_update?: boolean;
+    manual_download?: boolean;
+  };
+};
+
+type AutoupdateDismissedEvent = {
+  name: 'Autoupdate Dismissed';
+  payload: { update_version: string };
+};
+
+type ThemeChangedEvent = {
+  name: 'Theme Changed';
+  payload: { theme: 'DARK' | 'LIGHT' | 'OS_THEME' };
+};
+
+type KeytarSecretsMigrationFailedEvent = {
+  name: 'Keytar Secrets Migration Failed';
+  payload: { num_saved_connections: number; num_failed_connections: number };
+};
+
+type ConnectionExportedEvent = {
+  name: 'Connection Exported';
+  payload: { count: number };
+};
+
+type ConnectionImportedEvent = {
+  name: 'Connection Imported';
+  payload: { count: any };
+};
+
+type SwitchViewTypeEvent = ConnectionScoped<{
+  name: 'Switch View Type';
+  payload: { view_type: any; item_type: 'collection' | 'database' };
+}>;
+
+type CollectionCreatedEvent = ConnectionScoped<{
+  name: 'Collection Created';
+  payload: {
+    is_capped: boolean;
+    has_collation: boolean;
+    is_timeseries: boolean;
+    is_clustered: boolean;
+    is_fle2: boolean;
+    expires: boolean;
+  };
+}>;
+
+type DatabaseCreatedEvent = ConnectionScoped<{
+  name: 'Database Created';
+  payload: {
+    is_capped: boolean;
+    has_collation: boolean;
+    is_timeseries: boolean;
+    is_clustered: boolean;
+    is_fle2: boolean;
+    expires: boolean;
+  };
+}>;
```

Comment:
Honest Q: Who's the intended target audience for the JSDoc output, and would they maybe also be fine with this file? This is quite clean and readable already imo

Reply:
I would imagine that being ok yeah. Is us, analytics folks and product. I think that the comments would be enough to read. The main effort remaining would really be adding the missing documentation. I've grabbed what i could from the existing tracking plan, but it lacks lots of events :(.